flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskManagerRunner.java

```diff
@@ -293,30 +294,32 @@ public static void main(String[] args) throws Exception {
 			LOG.info("Cannot determine the maximum number of open file descriptors");
 		}
 
-		runTaskManagerSecurely(args, ResourceID.generate());
+		runTaskManagerSecurely(args, null);
 	}
 
 	public static Configuration loadConfiguration(String[] args) throws FlinkParseException {
 		return ConfigurationParserUtils.loadCommonConfiguration(args, TaskManagerRunner.class.getSimpleName());
 	}
 
-	public static void runTaskManager(Configuration configuration, ResourceID resourceId, PluginManager pluginManager) throws Exception {
-		final TaskManagerRunner taskManagerRunner = new TaskManagerRunner(configuration, resourceId, pluginManager);
+	public static void runTaskManager(Configuration configuration, PluginManager pluginManager) throws Exception {
+		final TaskManagerRunner taskManagerRunner = new TaskManagerRunner(configuration, getTaskManagerResourceID(configuration), pluginManager);
 
 		taskManagerRunner.start();
 	}
 
-	public static void runTaskManagerSecurely(String[] args, ResourceID resourceID) {
+	public static void runTaskManagerSecurely(String[] args, String resourceID) {
 		try {
 			final Configuration configuration = loadConfiguration(args);
-
+			if (resourceID != null) {
+				configuration.set(TaskManagerOptions.TASK_MANAGER_RESOURCE_ID, resourceID);
+			}
```

Comment:
I think it would be good if setting the `TASK_MANAGER_RESOURCE_ID` happens on the `ResourceManager` side.